src/SSHDebugPS/VSOperationWaiter.cs

```diff
@@ -30,39 +22,21 @@ internal static class VSOperationWaiter
         /// <exception cref="OperationCanceledException">Wait was canceled and 'throwOnCancel' is true</exception>
         public static bool Wait(string actionName, bool throwOnCancel, Action action)
         {
-            Task t = Task.Run(action);
-
-            DebugEngineHost.HostWaitLoop waiterImpl = null;
+            bool success = false;
 
-            try
-            {
-                waiterImpl = new DebugEngineHost.HostWaitLoop(actionName);
-            }
-            catch (FileNotFoundException)
-            {
-                // Visual Studio is not installed on this box
-            }
-
-            if (waiterImpl != null)
-            {
-                using (ManualResetEvent completeEvent = new ManualResetEvent(false))
+            ThreadHelper.JoinableTaskFactory.Run(
+                actionName,
+                async (progress, cancellationTokenSource) =>
                 {
-                    CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
-
-                    t.ContinueWith((System.Threading.Tasks.Task unused) => completeEvent.Set(), TaskContinuationOptions.ExecuteSynchronously);
-
-                    try
-                    {
-                        waiterImpl.Wait(completeEvent, cancellationTokenSource);
+                    try {
+                        await Task.Run(action, cancellationTokenSource);
                     }
                     catch (OperationCanceledException) // VS Wait dialog implementation always throws on cancel
                     {
                         if (throwOnCancel)
                         {
                             throw;
                         }
```

Comment:
nit: either add a 'when' handler and only catch the exception when `!throwOnCancel`, or remove this and let the below code throw.